ocaml/xapi/xapi_vif.ml

```diff
@@ -140,13 +142,15 @@ let set_ipv6_allowed ~__context ~self ~value =
   let setified_value = Listext.setify value in
   change_locking_config ~__context ~self ~licence_check:(setified_value <> [])
     (fun () ->
-      List.iter (Helpers.assert_is_valid_ip `ipv6 "ipv6_allowed") setified_value ;
```

Comment:
Is `setified_value` as stupid name? This is a set of value, i.e., is contains no duplicates. The name should be something like `addr_set` or similar. I understand this is existing code.

Reply:
I can change the names if you prefer.

Reply:
In general, I'd prefer using the ipaddr instead of this ad-hoc representation of IPs and CIDRs, it offers a sets of (IPV4)[https://ocaml.org/p/ipaddr/5.3.1/doc/Ipaddr/V4/Set/index.html] IPs, [IPV6](https://ocaml.org/p/ipaddr/5.3.1/doc/Ipaddr/V6/Set/index.html), [both](https://ocaml.org/p/ipaddr/5.3.1/doc/Ipaddr/Map/index.html), as well as CIDRs for either (called Prefix in the library). But this is for another PR